src/chksum.c

```diff
@@ -15,10 +15,42 @@
 #include "util.h"
 #include "chksum.h"
 
+#ifdef WITH_OPENSSL
+
+#include <openssl/md5.h>
+#include <openssl/sha.h>
+
+typedef SHA_CTX SHA1_CTX;
+typedef SHA256_CTX SHA224_CTX;
+typedef SHA512_CTX SHA384_CTX;
```

Comment:
Probably a noob question, why these defines?

Reply:
These types are defined in the internal implementation of hashes but not in OpenSSL. 
By defining these types, I can simple replace the internal implementation with OpenSSL (I have all needed types now). There is no need to additional code changes.

Reply:
I see, libsolv actually has the same typedefs. A short explanatory comment might be helpful here...

Otherwise, patch LGTM.